### ISSUE TITLE
Right click doesn't seem to work

### ISSUE THREAD

**Issue by briancappello** [plaidchat/plaidchat#20]

**Comment by twolfson** [comment]
By default, `nw.js` doesn't have any context menu enabled. This is to allow apps to customize the experience as much as possible.

https://github.com/nwjs/nw.js/issues/1861

What would you expect to be present in the context menu (e.g. copy selection)?

**Comment by garagatyi** [comment]
- copy
- search in google

**Comment by twolfson** [comment]
I will add to this list as well:

**When nothing is selected and right clicked on non-input element:**

- Select all (but that might be unnecessary)
    - Maybe we can do something Slack specific instead?
- Inspect element?

**When nothing is selected and right clicked on input element:**

- Select all
- Undo
- Redo
- Cut (greyed out)
- Copy (greyed out)
- Paste
- Delete (greyed out)

**When something is selected and right clicked on non-input element:**

- Select all?
- Copy
- Search Google for "{{selection}}"
- Inspect element?

**When something is selected and right clicked on input element:**

- Select all
- Undo
- Redo
- Cut
- Copy
- Paste
- Delete
- Search Google for "{{selection}}"

**Special case when selection matches a URL:**

- Open link in new window

@wlaurance Thoughts on these features and how necessary/unnecessary they are in `slack-for-linux`?

**Comment by garagatyi** [comment]
@twolfson +100 for this feature

Also would be helpful to add 
*When nothing is selected and right clicked on input element:*
- Paste as comment
*When something is selected and right clicked on non-input element:*
- Paste as comment
- Paste
*When something is selected and right clicked on input element:*
- Paste as comment

Also +1 for 
- When something is selected and shift + middle click put selection as comment

**Comment by twolfson** [comment]
Can you elaborate on "Paste as comment" and "Put selection as comment"? I am not sure what you mean by those.

**Comment by garagatyi** [comment]
in Slack you can format text as comment
> some text in one line
>>> some multiline text
Give it a try, very useful. 
I propose use > comment

**Comment by twolfson** [comment]
Ah, k. I think that will be tougher since that only applies to the chat input box (e.g. you don't want to post a comment in the search box). Since that is a more one-off case, we should consider that via a follow-up issue rather than as part of the initial implementation.

We are still waiting on @wlaurance for feedback on the rest of these options.

**Comment by mathieumg** [comment]
When nothing is selected and right clicked on a link (`<a href="...">...</a>`) element: 

* Open URL (optional since it's the primary click action)
* Copy link address

**Comment by justinnichols** [comment]
Just an FYI, I'm working in my fork on this issue.  I plan to start small with just the selectAll/cut/copy/paste/delete functions inside the textarea and then expand functionality from there.

**Comment by mathieumg** [comment]
@justinnichols A bit old (2.11 iirc), but if you need a starting point: https://github.com/mathieumg/plaidchat/tree/rightclick_menu

**Comment by justinnichols** [comment]
@mathieumg Thanks for the help, though I went in a different direction.  Hope it fits in well as I'm not too versed with React/Flux.

**Comment by twolfson** [comment]
Most of these have been implemented by @justinnichols in #136 which is now live in `v2.14.0`. Here are the remaining ones (that I can think of/listed in #136):

- [ ] Select All
- [ ] Undo
- [ ] Redo

**Comment by justinnichols** [comment]
Select All is there for the textarea, but not for any other elements (it was still up in the air as to whether to create that for non-input elements so I left it off).

**Comment by twolfson** [comment]
Ah, I must have missed it during review. Yea, we should probably add a "Select All" for non `textarea/input[type=text]` elements. Typically those select all the page contents.

Although, now that I think about it as a "Why would I want that". I don't think it's too applicable with respect to Slack. Removing that as a request =)
